ctapipe/io/datawriter.py

```diff
@@ -224,7 +241,14 @@ def __call__(self, event: ArrayEventContainer):
             )
 
         # write telescope event data
-        self._write_telescope_events(self._writer, event)
+        self._write_dl1_telescope_events(self._writer, event)
```

Comment:
Can we make these methods public? I don't really see a reason why they should be private and it might make using the writer to write specific thinks easier for users

Reply:
I don't think that would break anything, but it may be  confusing since it is a lower-level API function than `write()`.  Normally the user should just call `write(event)` and everything is handled automatically, and if they want something different there should be an option.  If they mix calling write() with calling these lower-level write functions  explicitly, it would lead to duplicated events, but perhaps that is ok.   

If we think the user wants more control, perhaps we should either add more options, or some *hook* functions that can be overridden?  Anyhow, i'm ok making these public, but I think that's better to do in another PR afterward.

Reply:
I was mainly thinking about inheriting from `DL1DataWriter` and overriding these methods with custom implementations. But since the underscore is just convention, I don't have a strong opinion.